components/game_table/commons/game_table_character.lua

```diff
@@ -0,0 +1,376 @@
+	})
+
+	local conditions = ConditionTree(BooleanOperator.any)
+	Array.forEach(lpdbData, function (game)
+		conditions:add(ConditionNode(ColumnName('match2id'), Comparator.eq, game.match2id))
+	end)
+
+	return conditions:toString()
+end
+
+---@param game match2game
+---@return integer, integer
+function CharacterGameTable:getCharacterPick(game)
+	if Logic.isEmpty(game.extradata) then
+		return CHARACTER_NOT_FOUND, CHARACTER_NOT_FOUND
+	end
+
+	local pickedByOpponent = CHARACTER_NOT_FOUND
+	local pickedByPlayer = CHARACTER_NOT_FOUND
+	---@param opponentIndex number
+	---@param playerIndex number
+	local findPick = function (opponentIndex, playerIndex)
+		if game.extradata[self:getCharacterKey(opponentIndex, playerIndex)] == self.args.character then
+			pickedByOpponent = opponentIndex
+			pickedByPlayer = playerIndex
+		end
+	end
+
+	self:_applyFuncToOpponentPlayers(1, findPick)
+	if pickedByOpponent == CHARACTER_NOT_FOUND then
+		self:_applyFuncToOpponentPlayers(2, findPick)
+	end
+
+	return pickedByOpponent, pickedByPlayer
+end
+
+---@param game match2game
+---@return match2game?
+function CharacterGameTable:gameFromRecord(game)
+	local gameRecord = GameTable.gameFromRecord(self, game)
+	if not gameRecord then
+		return nil
+	end
+
+	local pickedBy = self:getCharacterPick(game)
+	gameRecord.extradata.pickedBy = pickedBy
+	return pickedBy ~= CHARACTER_NOT_FOUND and gameRecord or nil
+end
+
+---@param record table
+---@return MatchTableMatchResult?
+function CharacterGameTable:resultFromRecord(record)
+	return {
+		opponent = record.match2opponents[1],
+		vs = record.match2opponents[2],
+		winner = tonumber(record.winner),
+		resultType = record.resultType,
+		countGames = true,
+	}
+end
+
+---@return {games: {w: number, d: number, l: number}}
+function CharacterGameTable:statsFromMatches()
+	local totalGames = {w = 0, d = 0, l = 0}
+
+	Array.forEach(self.matches, function(match)
+		---@cast match GameTableMatch
+		Array.forEach(match.games, function (game, index)
+			local winner = tonumber(game.winner)
+
+			if game.resulttype == DRAW then
+				totalGames.d = totalGames.d + 1
+			elseif game.extradata.pickedBy == winner then
+				totalGames.w = totalGames.w + 1
+			else
+				totalGames.l = totalGames.l + 1
+			end
+		end)
+	end)
+
+	return {
+		games = totalGames,
+	}
+end
+
+---@return Html
+function CharacterGameTable:headerRow()
+	local makeHeaderCell = function(text, width)
+		return mw.html.create('th'):css('max-width', width):node(text)
+	end
+
+	local nodes = Array.append({},
+		makeHeaderCell('Date', '100px'),
+		makeHeaderCell(self.config.showTier and 'Tier', '70px') or nil,
+		makeHeaderCell(nil, '25px'):addClass('unsortable'),
+		makeHeaderCell('Tournament')
+	)
+	if self.isCharacterTable then
+		nodes = Array.append(nodes,
+			makeHeaderCell('Team'):addClass('unsortable'),
+			makeHeaderCell(nil, '80px'),
+			makeHeaderCell('Score'),
+			makeHeaderCell(nil, '80px'),
+			makeHeaderCell('vs. Team'):addClass('unsortable')
```

Comment:
it isn't always teams ...
(e.g. warcraft is a mostly 1v1 based wiki which has heroes too)